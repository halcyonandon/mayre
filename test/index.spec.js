import React from 'react'
import { shallow } from 'enzyme'

// import Mayre from '../build/bundle.min'
const Mayre = (props) => {
  return props.when ? <props.of {...props.with} /> : null
}

const Tryout = (props) => <p id='tryout'>{props.name}</p>

describe('Mayre suite', () => {
  fit('should render any component through `of`', () => {
    const wrapper = shallow(<Mayre of={Tryout} />)
    expect(wrapper.find(Tryout)).toBe(true)
  })

  it('should render any component with props through `with`', () => {
    expect(<Mayre of={Tryout} with={{ name: 'Gamora' }}/>).toMatchSnapshot()
  })

  it('should render any component with a positive condition through `when`', () => {
    expect(<Mayre of={Tryout} when={5 > 10} />).toMatchSnapshot()
  })

  it('should not render any component with a negative condition through `when`', () => {
    expect(<Mayre of={Tryout} when={5 < 10} />).toMatchSnapshot()
  })

  it('should choose if render any component with a function condition through `when`', () => {
    expect(<Mayre of={Tryout} when={() => true} />).toMatchSnapshot()
  })
})
