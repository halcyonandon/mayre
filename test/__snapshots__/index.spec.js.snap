// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mayre suite should choose if render any component with a function condition through \`when\` 1`] = `
<Mayre
  of={[Function]}
  when={[Function]}
/>
`;

exports[`Mayre suite should not render any component with a negative condition through \`when\` 1`] = `
<Mayre
  of={[Function]}
  when={true}
/>
`;

exports[`Mayre suite should render any component through \`of\` 1`] = `
<Mayre
  of={[Function]}
/>
`;

exports[`Mayre suite should render any component with a positive condition through \`when\` 1`] = `
<Mayre
  of={[Function]}
  when={false}
/>
`;

exports[`Mayre suite should render any component with props through \`with\` 1`] = `
<Mayre
  of={[Function]}
  with={
    Object {
      "name": "Gamora",
    }
  }
/>
`;
